dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/controller/RequestController.kt

```diff
@@ -77,37 +80,53 @@ class RequestController(
     }
 
     override fun getDataRequests(
-        dataType: DataTypeEnum?,
+        dataType: Set<DataTypeEnum>?,
         userId: String?,
-        requestStatus: RequestStatus?,
-        accessStatus: AccessStatus?,
+        emailAddress: String?,
+        requestStatus: Set<RequestStatus>?,
+        accessStatus: Set<AccessStatus>?,
         reportingPeriod: String?,
         datalandCompanyId: String?,
         chunkSize: Int,
         chunkIndex: Int,
     ): ResponseEntity<List<ExtendedStoredDataRequest>> {
-        val currentUserId = DatalandAuthentication.fromContext().userId
-        val companyRoleAssignmentsOfCurrentUser =
-            companyRolesManager.getCompanyRoleAssignmentsByParameters(null, null, userId = currentUserId)
-        val filter = DataRequestsQueryFilter(
-            dataTypeFilter = dataType?.value ?: "",
-            userIdFilter = userId ?: "",
-            requestStatus = requestStatus?.name ?: "",
-            accessStatus = accessStatus?.name ?: "",
-            reportingPeriodFilter = reportingPeriod ?: "",
-            datalandCompanyIdFilter = datalandCompanyId ?: "",
+        val filter = DataRequestsFilter(
+            dataType, userId, emailAddress, datalandCompanyId, reportingPeriod, requestStatus, accessStatus,
         )
+
+        val authenticationContext = DatalandAuthentication.fromContext()
```

Comment:
Gibt es einen Grund das hier im Controller und nicht in dem Service zu machen?

Reply:
Ja, der CompanyRoleManager kann nicht in den DataRequestQueryManager importiert werden, da es sonst zu zyklischen Abhängigkeiten zwischen den Beans/Components kommt.

Reply:
![image](https://github.com/user-attachments/assets/3ddf47b9-9e71-4cd6-874a-28e5d270e761)
Ist das der Grund für die zyklische Abhängigkeit?

Reply:
Ja, scheint DataRequestQueryManager -> CompanyOwnershipAcceptedEmailMessageSender -> CompanyRolesManager zu sein

Reply:
Ok, mir ist auch keine Lösung eingefallen. Ich mache hier zu.